File: mRemoteV1/CS/Forms/OptionsPages/ThemePage.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using System.ComponentModel;
using mRemoteNG.My;
using mRemoteNG.Themes;


namespace mRemoteNG.Forms.OptionsPages
{
	public partial class ThemePage
		{
			public ThemePage()
			{
				InitializeComponent();
			}
public override string PageName
			{
				get
				{
					return Language.strOptionsTabTheme;
				}
				set
				{
				}
			}
			
			public override void ApplyLanguage()
			{
				base.ApplyLanguage();
				
				btnThemeDelete.Text = Language.strOptionsThemeButtonDelete;
				btnThemeNew.Text = Language.strOptionsThemeButtonNew;
			}
			
			public override void LoadSettings()
			{
				base.SaveSettings();
				
				_themeList = new BindingList<ThemeInfo>(ThemeManager.LoadThemes());
				cboTheme.DataSource = _themeList;
				cboTheme.SelectedItem = ThemeManager.ActiveTheme;
				cboTheme_SelectionChangeCommitted(this, new EventArgs());
				
				ThemePropertyGrid.PropertySort = PropertySort.Categorized;
				
				_originalTheme = ThemeManager.ActiveTheme;
			}
			
			public override void SaveSettings()
			{
				base.SaveSettings();
				
				ThemeManager.SaveThemes(_themeList);
				Settings.ThemeName = ThemeManager.ActiveTheme.Name;
			}
			
			public override void RevertSettings()
			{
				ThemeManager.ActiveTheme = _originalTheme;
			}
			
#region Private Fields
			private BindingList<ThemeInfo> _themeList;
			private ThemeInfo _originalTheme;
#endregion
			
#region Private Methods
#region Event Handlers
			public void cboTheme_DropDown(object sender, EventArgs e)
			{
				if (ThemeManager.ActiveTheme == ThemeManager.DefaultTheme)
				{
					return ;
				}
				ThemeManager.ActiveTheme.Name = cboTheme.Text;
			}
			
			public void cboTheme_SelectionChangeCommitted(object sender, EventArgs e)
			{
				if (cboTheme.SelectedItem == null)
				{
					cboTheme.SelectedItem = ThemeManager.DefaultTheme;
				}
				
				if (cboTheme.SelectedItem == ThemeManager.DefaultTheme)
				{
					cboTheme.DropDownStyle = ComboBoxStyle.DropDownList;
					btnThemeDelete.Enabled = false;
					ThemePropertyGrid.Enabled = false;
				}
				else
				{
					cboTheme.DropDownStyle = ComboBoxStyle.DropDown;
					btnThemeDelete.Enabled = true;
					ThemePropertyGrid.Enabled = true;
				}
				
				ThemeManager.ActiveTheme = cboTheme.SelectedItem;
				ThemePropertyGrid.SelectedObject = ThemeManager.ActiveTheme;
				ThemePropertyGrid.Refresh();
			}
			
			public void btnThemeNew_Click(object sender, EventArgs e)
			{
				ThemeInfo newTheme = ThemeManager.ActiveTheme.Clone();
				newTheme.Name = Language.strUnnamedTheme;
				
				_themeList.Add(newTheme);
				
				cboTheme.SelectedItem = newTheme;
				cboTheme_SelectionChangeCommitted(this, new EventArgs());
				
				cboTheme.Focus();
			}
			
			public void btnThemeDelete_Click(object sender, EventArgs e)
			{
				ThemeInfo theme = cboTheme.SelectedItem;
				if (theme == null)
				{
					return ;
				}
				
				_themeList.Remove(theme);
				
				cboTheme.SelectedItem = ThemeManager.DefaultTheme;
				cboTheme_SelectionChangeCommitted(this, new EventArgs());
			}
#endregion
#endregion
		}
}

File: mRemoteV1/Forms/OptionsPages/ThemePage.vb
Imports System.ComponentModel
Imports mRemote3G.Themes

Namespace Forms.OptionsPages
    Public Class ThemePage
        Public Overrides Property PageName As String
            Get
                Return Language.Language.strOptionsTabTheme
            End Get
            Set
            End Set
        End Property

        Public Overrides Sub ApplyLanguage()
            MyBase.ApplyLanguage()

            btnThemeDelete.Text = Language.Language.strOptionsThemeButtonDelete
            btnThemeNew.Text = Language.Language.strOptionsThemeButtonNew
        End Sub

        Public Overrides Sub LoadSettings()
            MyBase.SaveSettings()

            _themeList = New BindingList(Of ThemeInfo)(ThemeManager.LoadThemes())
            cboTheme.DataSource = _themeList
            cboTheme.SelectedItem = ThemeManager.ActiveTheme
            cboTheme_SelectionChangeCommitted(Me, New EventArgs())

            ThemePropertyGrid.PropertySort = PropertySort.Categorized

            _originalTheme = ThemeManager.ActiveTheme
        End Sub

        Public Overrides Sub SaveSettings()
            MyBase.SaveSettings()

            ThemeManager.SaveThemes(_themeList)
            My.Settings.ThemeName = ThemeManager.ActiveTheme.Name
        End Sub

        Public Overrides Sub RevertSettings()
            ThemeManager.ActiveTheme = _originalTheme
        End Sub

#Region "Private Fields"

        Private _themeList As BindingList(Of ThemeInfo)
        Private _originalTheme As ThemeInfo

#End Region

#Region "Private Methods"

#Region "Event Handlers"

        Private Sub cboTheme_DropDown(sender As Object, e As EventArgs) Handles cboTheme.DropDown
            If ThemeManager.ActiveTheme Is ThemeManager.DefaultTheme Then Return
            ThemeManager.ActiveTheme.Name = cboTheme.Text
        End Sub

        Private Sub cboTheme_SelectionChangeCommitted(sender As Object, e As EventArgs) _
            Handles cboTheme.SelectionChangeCommitted
            If cboTheme.SelectedItem Is Nothing Then cboTheme.SelectedItem = ThemeManager.DefaultTheme

            If cboTheme.SelectedItem Is ThemeManager.DefaultTheme Then
                cboTheme.DropDownStyle = ComboBoxStyle.DropDownList
                btnThemeDelete.Enabled = False
                ThemePropertyGrid.Enabled = False
            Else
                cboTheme.DropDownStyle = ComboBoxStyle.DropDown
                btnThemeDelete.Enabled = True
                ThemePropertyGrid.Enabled = True
            End If

            ThemeManager.ActiveTheme = cboTheme.SelectedItem
            ThemePropertyGrid.SelectedObject = ThemeManager.ActiveTheme
            ThemePropertyGrid.Refresh()
        End Sub

        Private Sub btnThemeNew_Click(sender As Object, e As EventArgs) Handles btnThemeNew.Click
            Dim newTheme As ThemeInfo = ThemeManager.ActiveTheme.Clone()
            newTheme.Name = Language.Language.strUnnamedTheme

            _themeList.Add(newTheme)

            cboTheme.SelectedItem = newTheme
            cboTheme_SelectionChangeCommitted(Me, New EventArgs())

            cboTheme.Focus()
        End Sub

        Private Sub btnThemeDelete_Click(sender As Object, e As EventArgs) Handles btnThemeDelete.Click
            Dim theme As ThemeInfo = cboTheme.SelectedItem
            If theme Is Nothing Then Return

            _themeList.Remove(theme)

            cboTheme.SelectedItem = ThemeManager.DefaultTheme
            cboTheme_SelectionChangeCommitted(Me, New EventArgs())
        End Sub

#End Region

#End Region
    End Class
End Namespace